### PR TITLE
chore: Upgrade to Kubernetes v0.23. Fixes #6750

### LINKED ISSUE #6750: Argo Client does not work with newer k8s client versions (>=v0.21.0)
## Summary 
I am using the go-client for argo and here's how I defined it:

```
	opts := apiclient.Opts{
		ArgoServerOpts: apiclient.ArgoServerOpts{
			URL:                argoURI,
			Path:               argoPath,
			Secure:             true,
			InsecureSkipVerify: false,
			HTTP1:              true,
		},
		AuthSupplier: func() string {
			return ""
		},
	}
	argoCtx, argo, err := apiclient.NewClientFromOpts(opts)
	if err != nil {
		return "", err
	}
```

My go.mod file:
```
module ...

go 1.15

require (
	github.com/argoproj/argo-workflows/v3 v3.0.0
	github.com/go-logr/logr v0.4.0
	github.com/go-yaml/yaml v2.1.0+incompatible
	github.com/onsi/ginkgo v1.14.2
	github.com/onsi/gomega v1.10.4
	gitlab-odx.oracledx.com/cx-sales/code/titan/titan-core v0.2.0
	k8s.io/apimachinery v0.21.0
	k8s.io/client-go v0.21.0
	sigs.k8s.io/controller-runtime v0.7.2
)

```

What happened/what you expected to happen?
I need to use k8s go client version >= v0.21.0 but it seems the latest argo client is incompatible with the newer k8s versions. So

Running `go vet ./... `gives an error:

```# github.com/argoproj/argo-workflows/v3/util/kubeconfig
..\..\..\..\..\go\pkg\mod\github.com\argoproj\argo-workflows\v3@v3.1.12\util\kubeconfig\kubeconfig.go:147:37: not enough arguments in call 
to exec.GetAuthenticator
        have (*api.ExecConfig)
        want (*api.ExecConfig, *clientauthentication.Cluster)
```

Clearly, the exec.GetAuthenticator function requires two arguments in the newer k8s client versions. Can a bug fix be released for this purpose? Alternately, how should I define my apiClient to be compatible with the latest k8s client? Would appreciate any pointers!

## Diagnostics

What Kubernetes provider are you using? 
Terraform

What version of Argo Workflows are you running? 
v0.3.12

What executor are you running? Docker/K8SAPI/Kubelet/PNS/Emissary
Docker

Did this work in a previous version? I.e. is it a regression?
No

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by alexec**
Signed-off-by: Alex Collins <alex_collins@intuit.com>

Fixes #6750

### CODE REVIEW

**Comment by alexec** [comment]
Blocked by https://github.com/argoproj/argo-events/pull/1610

**Comment by JPZ13** [comment]
The hack secondary swagger gen file might need to be changes. Here's what I wound up doing:
https://github.com/pipekit/argo-workflows/blob/upgrade-openapi/hack/swagger/secondaryswaggergen.go#L9
https://github.com/pipekit/argo-workflows/blob/upgrade-openapi/hack/swagger/secondaryswaggergen.go#L26

File: docs/cli/argo_resubmit.md [hunk 1]
```diff
@@ -65,6 +65,7 @@ argo resubmit [WORKFLOW...] [flags]
   -s, --argo-server host:port          API server host:port. e.g. localhost:2746. Defaults to the ARGO_SERVER environment variable.
       --as string                      Username to impersonate for the operation
       --as-group stringArray           Group to impersonate for the operation, this flag can be repeated to specify multiple groups.
+      --as-uid string                  UID to impersonate for the operation
```
Comment (terrytangyuan) [inline docs/cli/argo_resubmit.md:68]: Is this change intended in this PR?

Comment (alexec) [reply docs/cli/argo_resubmit.md:68]: it is a consquence of the upgrade

File: Makefile [hunk 2]
```diff
@@ -539,7 +539,7 @@ pkg/apis/workflow/v1alpha1/zz_generated.deepcopy.go: $(TYPES)
 
 dist/kubernetes.swagger.json:
 	@mkdir -p dist
-	./hack/recurl.sh dist/kubernetes.swagger.json https://raw.githubusercontent.com/kubernetes/kubernetes/v1.17.5/api/openapi-spec/swagger.json
+	./hack/recurl.sh dist/kubernetes.swagger.json https://raw.githubusercontent.com/kubernetes/kubernetes/v1.23.3/api/openapi-spec/swagger.json
```
[the remark this answers is not in the document]
Comment (alexec) [reply Makefile:542]: yes, I'll do this

File: docs/cli/argo_resubmit.md (hunk 1 above)
Comment (terrytangyuan) [reply docs/cli/argo_resubmit.md:68]: Anything that needs to be documented in upgrading guide?

File: Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexec) [reply Makefile:542]: having reviewed .4 fixes and considered the extra work of upgrading dataflow and events, I think we should leave this for now

**Review by terrytangyuan — APPROVED** [review]
LGTM!

**Pull request merged**